### ISSUE TITLE
URL is not shown in the address bar in vertical tab view  if title bar is hidden in windows snap windows layout

### ISSUE THREAD

**Issue by MadhaviSeelam** [brave/brave-browser#29874]
## Description 

URL of a visited site is not visible in the address bar in split screen mode and in vertical tab view mode and title bar is hidden.  (browser window dimensions are 756 x 676px).  Address bar is adjusted when title bar is shown and in horizontal view to show complete URL as expected. Reproduced in 1.52.57 (Nightly)

## Steps to Reproduce

   1. Install 1.51.95
   2. launch Brave
   3. visit abcnews.com
   4. enable vertical tabs in brave://flags
   5. enable `Use vertical tabs` in context menu
   6. enable split screen mode (Win + left/right arrow)
   

## Actual result:

<img width="652" alt="image" src="https://user-images.githubusercontent.com/98358127/233220948-e2272391-0cfb-434e-aac7-010ff99ca5df.png">

## Expected result:
Address bar should be adjusted to show complete or partial address? 

## Reproduces how often: 

Easily

## Brave version (brave://version info)

```
1.51.95 Chromium: 112.0.5615.121 (Official Build) beta (64-bit)
--
Revision | 39cc4e45904ae9f1741c4fbaa866e629c96f2268-refs/branch-heads/5615_51@{#8}
OS | Windows 11 Version 22H2 (Build 22621.1555)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release?  N/A
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:
@rebron @sangwoo108 
cc: @brave/qa-team

**Comment by sangwoo108** [comment]
I believe this could be reproducible when a browser window is foreced to have narrow width with horizontal tab strip as well. 

<img width="517" alt="image" src="https://user-images.githubusercontent.com/5474642/233226588-0d9a931c-8449-4bcc-8fd4-ed7a804a2b04.png">


As when hiding title bar, toolbar run out of enough space as caption button's will take space, it's easier to happen. We might want to label this as `feature-request` and `feature/tabs-bar` instead.

**Comment by petemill** [comment]
Yeah this shouldn't be blocking, but I think we can ask @aguscruiz if there's a solution planned when the window gets this narrow...

**Comment by aguscruiz** [comment]
I have nothing planned yet for this scenario, but I can keep it in mind for future improvements!
